src/whylogs/core/statistics/constraints.py

```diff
@@ -251,13 +259,26 @@ def name(self):
         if self.op == Op.BTWN:
             lower_target = self.value if self.value is not None else self.second_field
             upper_target = self.upper_value if self.upper_value is not None else self.third_field
-            return self._name if self._name is not None else f"summary {self.first_field} {Op.Name(self.op)} {lower_target} and {upper_target}"
+            field_name = self.first_field
+            if self.first_field == "quantile":
+                field_name = f"{self.first_field} {self.quantile_value}"
+            return self._name if self._name is not None else f"summary {field_name} {Op.Name(self.op)} {lower_target} and {upper_target}"
 
         return self._name if self._name is not None else f"summary {self.first_field} {Op.Name(self.op)} {self.value}/{self.second_field}"
 
-    def update(self, summ: NumberSummary) -> bool:
+    def update(self, update_dict: dict) -> bool:
         self.total += 1
-        if not self.func(summ):
+        summ = update_dict["number_summary"]
+        column_number_kll_sketch = update_dict["number_kll_sketch"]
+
+        if self.first_field == "quantile":
+            quantile_summary = quantiles_from_sketch(column_number_kll_sketch, [self.quantile_value])
+            quantile_val = type("Object", (), {self.first_field: quantile_summary.quantile_values[0]})
```

Comment:
Can you explain this some? Do we need a wrapper or utility function for the quantile values?

Reply:
The user specifies the quantile value for which they want to apply the constraint. This value is passed in the constructor of the SummaryConstraint in the quantile_value argument. The quantiles_from_sketch method calculates the quantiles and their value based on the number summary kll sketch, and the passed quantiles (list) as a second parameter. The first_first field here is only used to check if this is a quantile constraint and then calculate the quantile value (one quantile), which is then passed as an anonymous object with an attribute quantile = the_quantile_value. This is because the implementation of the summary operators requires an object from which it gets the corresponding field (field_name) using the operator. 

We could use a utility function, it would make the interface clearer.

Reply:
yes a utility function here would be better to encapsulate that interface expectation/wrapping. And/or add a comment hinting at the reason we need to do this in the code so that reading the code is easier to understand.